WordPress/gutenberg
# Eslint Plugin: Change plugin dependencies to peer dependencies

Fixes #39810



## What?
Change dependencies to match recommendations of shareable configs in Eslint. For reference https://eslint.org/docs/latest/developer-guide/shareable-configs#publishing-a-shareable-config

## Why?
Avoid potential versioning conflicts when package is used in other projects, or if Gutenberg introduces new packages related to eslint.

## How?
Moves the eslint plugin dependencies into peer dependencies using the >= range operator.

## Testing Instructions

1. Clear your `node_modules` folder.
2. Run `npm i` to install deps.
3. Run `npm run lint-js` to ensure that all plugins for eslint are still loading correctly, you should see warnings still.

## Screenshots or screencast 
Not applicable
